cobc/parser.y

```diff
@@ -15391,6 +15391,15 @@ stop_statement:
 	check_unreached = 1;
 	cobc_cs_check = 0;
   }
+| STOP ERROR /* GCOS */
+  {
+	begin_statement ("STOP ERROR", 0);
+	cb_verify (cb_stop_error, "STOP ERROR");
+	cobc_cs_check = CB_CS_STOP;
+	cb_emit_stop_run (cb_int1);
```

Comment:
The GCOS docs say

> The ERROR phrase has been implemented for debugging purpose. If used, the
execution of the run unit aborts and control is transferred to the operating system. It
is the way, in case of malfunction, to get a dump where all elements (hardware
registers, etc.) remain unchanged.

So instead of a "normal" `STOP RUN RETURNING 1` it should likely trigger the dump feature, shouldn't it?
That would be triggered by any exception, we could add a new fatal one `EC-PROGRAM-STOP-ERROR` or `EC-IMP-STOP-ERROR` and let it be triggered there. This would automatically return with 1 but before trigger exception handlers (both possibly registered ones and the internal one which triggers the dump for all modules compiled with that feature).

Reply:
Trying to add a `EC-IMP-STOP-ERROR` exception (set to fatal in exception.def).
Not sure to know how to actually raise it. I use `cob_set_exception (COB_EC_IMP_STOP_ERROR)`, but this only sets the last exception.

Would just using a simple `cob_runtime_error ("STOP ERROR")` instead of using exceptions do it ?

Reply:
I agree that adding the exception alone won't help. The most reasonable thing to do is to emit `cob_runtime_error ("STOP ERROR");` and `cob_stop_run(1);`.

Note: when there are no more "context sensitive" tokens to follow on (which does not seem the case here) you don't need to set the context-sensitive words check `cobc_cs_check`, so that line can likely be dropped.